lib/src/widgets/game_status.dart

```diff
@@ -11,15 +12,32 @@ class GameStatusWidget extends StatelessWidget {
     required this.currentTurn,
     required this.userColor,
     required this.gameCode,
+    required this.turnOrder,
+    required this.turnNumber,
   }) : super(key: key);
 
   final List<Player> players;
   final PlayerColor currentTurn;
   final PlayerColor userColor;
   final String gameCode;
+  final List<PlayerColor> turnOrder;
+  final int turnNumber;
+  
 
   @override
   Widget build(BuildContext context) {
+  final remaingTilesMap = <PlayerColor,int> {};
+
+    for (var i = 0; i<turnOrder.length; i++) {
+      var playerColor = turnOrder[i];
+      remaingTilesMap[playerColor] = GameState.tilesAvailablePerPlayer;
+    }
+
+    for (var i = 0; i<turnNumber; i++ ) {
+      var playerColor = turnOrder [(i) % turnOrder.length];
+      remaingTilesMap[playerColor] = remaingTilesMap[playerColor]! - 1;
+    }
```

Comment:
This code looks good, but I think there's a way you can optimize it to make it loop fewer times.

Can you think of a way to make it so you don't have to loop through `turnNumber` of times and get the same result?